mec/mec1501/component/spi_periph.h

```diff
@@ -0,0 +1,341 @@
+#define MCHP_SPIP_SPI_IEN_TXF_OVF	(1ul << 26)
+#define MCHP_SPIP_SPI_IEN_RXF_UNF	(1ul << 27)
+#define MCHP_SPIP_SPI_IEN_RXF_OVF	(1ul << 28)
+
+/* SPIP EC Interrupt Enable register */
+#define MCHP_SPIP_EC_IEN_REG_OFS	0x10
+#define MCHP_SPIP_EC_IEN_REG_MASK	0x1FFFEF7Bul
+#define MCHP_SPIP_EC_IEN_MWD		(1ul << 0)
+#define MCHP_SPIP_EC_IEN_MRD		(1ul << 1)
+#define MCHP_SPIP_EC_IEN_MWB		(1ul << 3)
+#define MCHP_SPIP_EC_IEN_MRB		(1ul << 4)
+#define MCHP_SPIP_EC_IEN_SRTB		(1ul << 5)
+#define MCHP_SPIP_EC_IEN_PHI_REQ	(1ul << 6)
+#define MCHP_SPIP_EC_IEN_RXFE		(1ul << 8)
+#define MCHP_SPIP_EC_IEN_RXFF		(1ul << 9)
+#define MCHP_SPIP_EC_IEN_TXFE		(1ul << 10)
+#define MCHP_SPIP_EC_IEN_TXFF		(1ul << 11)
+#define MCHP_SPIP_EC_IEN_SCCE		(1ul << 13)
+#define MCHP_SPIP_EC_IEN_IBF		(1ul << 14)
+#define MCHP_SPIP_EC_IEN_OBF		(1ul << 15)
+#define MCHP_SPIP_EC_IEN_SPIMR		(1ul << 16)
+#define MCHP_SPIP_EC_IEN_RXF_RSTD	(1ul << 17)
+#define MCHP_SPIP_EC_IEN_TXF_RSTD	(1ul << 18)
+#define MCHP_SPIP_EC_IEN_LIM0		(1ul << 19)
+#define MCHP_SPIP_EC_IEN_LIM1		(1ul << 20)
+#define MCHP_SPIP_EC_IEN_ABERR		(1ul << 21)
+#define MCHP_SPIP_EC_IEN_UNDEF_CMD	(1ul << 22)
+#define MCHP_SPIP_EC_IEN_DVB		(1ul << 23)
+#define MCHP_SPIP_EC_IEN_RXF_SZ		(1ul << 24)
+#define MCHP_SPIP_EC_IEN_TXF_UNF	(1ul << 25)
+#define MCHP_SPIP_EC_IEN_TXF_OVF	(1ul << 26)
+#define MCHP_SPIP_EC_IEN_RXF_UNF	(1ul << 27)
+#define MCHP_SPIP_EC_IEN_RXF_OVF	(1ul << 28)
+
+/* SPIP Memory Config register */
+#define MCHP_SPIP_MCFG_REG_OFS		0x14
+#define MCHP_SPIP_MCFG_REG_MASK		0x03ul
+#define MCHP_SPIP_MCFG_BAR0_EN_POS	0
+#define MCHP_SPIP_MCFG_BAR0_EN		(1ul << 0)
+#define MCHP_SPIP_MCFG_BAR1_EN_POS	1
+#define MCHP_SPIP_MCFG_BAR1_EN		(1ul << 1)
+
+/* SPIP Memory Base Address 0 register */
+#define MCHP_SPIP_MBA0_REG_OFS		0x18
+#define MCHP_SPIP_MBA0_REG_MASK		0xFFFFFFFCul
+
+/* SPIP Memory Write Limit 0 register */
+#define MCHP_SPIP_MWLIM0_REG_OFS	0x1C
+#define MCHP_SPIP_MWLIM0_REG_MASK	0xFFFFFFFCul
+
+/* SPIP Memory Read Limit 0 register */
+#define MCHP_SPIP_MRLIM0_REG_OFS	0x20
+#define MCHP_SPIP_MRLIM0_REG_MASK	0xFFFFFFFCul
+
+/* SPIP Memory Base Address 1 register */
+#define MCHP_SPIP_MBA1_REG_OFS		0x24
+#define MCHP_SPIP_MBA1_REG_MASK		0xFFFFFFFCul
+
+/* SPIP Memory Write Limit 1 register */
+#define MCHP_SPIP_MWLIM1_REG_OFS	0x28
+#define MCHP_SPIP_MWLIM1_REG_MASK	0xFFFFFFFCul
+
+/* SPIP Memory Read Limit 1 register */
+#define MCHP_SPIP_MRLIM1_REG_OFS	0x2C
+#define MCHP_SPIP_MRLIM1_REG_MASK	0xFFFFFFFCul
+
+/* SPIP RX FIFO Host BAR register */
+#define MCHP_SPIP_RXFHB_REG_OFS	0x30
+#define MCHP_SPIP_RXFHB_REG_MASK	0xFFFFul
+
+/* SPIP RX FIFO Host BAR register */
+#define MCHP_SPIP_RXFBC_REG_OFS	0x34
+#define MCHP_SPIP_RXFBC_REG_MASK	0x7FFFul
+
+/* SPIP TX FIFO Host BAR register */
+#define MCHP_SPIP_TXFHB_REG_OFS	0x38
+#define MCHP_SPIP_TXFHB_REG_MASK	0xFFFFul
+
+/* SPIP TX FIFO Host BAR register */
+#define MCHP_SPIP_TXFBC_REG_OFS	0x3C
+#define MCHP_SPIP_TXFBC_REG_MASK	0x7FFFul
```

Comment:
Please use lower-case hex consistently

Reply:
I will be changing "ul" to "U".  Do you also want all hex digits to be one case?  0x3CU and 0x7FFFU or 0x3cU and 0x7fffU

Reply:
I much prefer u to U. Also all hex in lower-case as easier to read

So if you are required to add u, then 0xfffu

I would note that u seems silly for things which are not actually integers, but bit masks, etc. I am not sure if there guidance on that. Honestly I find the conventions here just add noise to the source code, but it seems to be agreed that we do this.

Reply:
@sjg20 maybe you want to revisit https://docs.zephyrproject.org/latest/contribute/index.html so we have a common ground for discussions.

Reply:
Converted hex digits to lower case.  Changed all "ul/UL" to "u".  Hope my python scripts found all of them and didn't corrupt the files.